server/service/client.go

```diff
@@ -520,10 +555,10 @@ func (c *Client) ApplyGroup(
 		tmMDMSettings := extractTmSpecsMDMCustomSettings(specs.Teams)
 
 		tmFileContents := make(map[string][]fleet.MDMProfileBatchPayload, len(tmMDMSettings))
-		for k, paths := range tmMDMSettings {
-			fileContents, err := getProfilesContents(baseDir, paths, opts.ExpandEnvConfigProfiles)
+		for k, profileSpecs := range tmMDMSettings {
+			fileContents, err := getProfilesContents(baseDir, profileSpecs.macos, profileSpecs.windows, opts.ExpandEnvConfigProfiles)
 			if err != nil {
-				return nil, err
+				return nil, fmt.Errorf("Team %s: %w", k, err) // TODO: add team name to error messages for other parts of the config
```

Comment:
Did you mean to add the team name as part of this PR?

Reply:
I meant it as a prompt to revisit it in the future so I've updated the comment.